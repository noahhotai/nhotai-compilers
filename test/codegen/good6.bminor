

the_array: array [3] integer  = {1, 2, 3};

the_string: string = "the_string\n";


main: function integer () = {

    j: string = the_string;
    print the_string, j;

    i:integer = 1;
    print i++, i;
}

