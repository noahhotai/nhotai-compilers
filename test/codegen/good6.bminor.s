.file "test/codegen/good6.bminor.s"
.data
the_array: 
.quad 1, 2, 3
.text
.global main
main:
PUSHQ %rbp
MOVQ %rsp, %rbp
PUSHQ %rdi
PUSHQ %rsi
PUSHQ %rdx
PUSHQ %rcx
PUSHQ %r8
PUSHQ %r9
MOVQ $1, %rbx
PUSHQ %rbx
MOVQ -56(%rbp), %rbx
INCQ -56(%rbp)
MOVQ %rbx, %rdi
CALL print_integer
MOVQ -56(%rbp), %rbx
MOVQ %rbx, %rdi
CALL print_integer
ADDQ $8, %rsp
POPQ %r9
POPQ %r8
POPQ %rcx
POPQ %rdx
POPQ %rsi
POPQ %rdi
POPQ %rbp
RET
