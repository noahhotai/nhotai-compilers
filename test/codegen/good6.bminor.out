Program parsed successfully.
the_string resolves to global 1
the_string resolves to global 1
j resolves to local 0
i resolves to local 1
i resolves to local 1
Program resolved successfully.
Program type checked successfully.
